lib/metadata.js

```diff
@@ -0,0 +1,209 @@
+                logger.debug(`In ${this.constructor.name}.getProperty() -> request.get().then()`);
+
+                return responseHandler.handleMetadataResponse(response).then((responseObj) => {
+                    return responseObj;
+                });
+            });
+        }
+
+
+        /**
+         * Updates the artifact's metadata. Only updates mutable items.
+         *
+         * @param {Object} metadata The data to add to the artifact's metadata.
+         * @return {Promise.<Object>}
+         */
+        updateAll(metadata) {
+            var options;
+
+            logger.debug(`In ${this.constructor.name}.updateAll()`);
+            logger.debug(`Metadata param: ${metadata}`);
+
+            if (!metadata) {
+                return Promise.reject("Must provide metadata to update.");
+            }
+
+            options = requestOptions.createOptions(this.authToken);
+            options.url = this.uri;
+            options.body = metadata;
+            logger.debug(`PUT request options: ${JSON.stringify(options, null, 4)}`);
+
+            return requestPromise.put(options).then((response) => {
+                logger.debug(`In ${this.constructor.name}.updateAll() -> request.put().then()`);
+
+                return responseHandler.handleMetadataResponse(response).then((responseObj) => {
+                    return responseObj;
+                });
+            });
+        }
+
+
+        /**
+         * Updates the artifact's metadata. Only updates mutable items.
+         * If the property doesn't exist, it will be created.
+         *
+         * @param {string} name The metadata property to update.
+         * @param {Object} property The metadata to update on the artifact.
+         * @return {Promise.<Object>}
+         */
+        updateProperty(name, property) {
+            var options;
+
+            logger.debug(`In ${this.constructor.name}.updateProperty()`);
+            logger.debug(`Name param: ${name}`);
+            logger.debug(`Property param: ${property}`);
+
+            if (!name) {
+                return Promise.reject("Must provide the name for the property to update.");
+            }
+
+            if (!property) {
+                return Promise.reject("Must provide metadata to update.");
+            }
+
+            options = requestOptions.createOptions(this.authToken);
+            options.url = `${this.uri}/${name}`;
+            options.body = property;
+            logger.debug(`PUT request options: ${JSON.stringify(options, null, 4)}`);
+
+            return requestPromise.put(options).then((response) => {
+                logger.debug(`In ${this.constructor.name}.updateProperty() -> request.put().then()`);
+
+                return responseHandler.handleMetadataResponse(response).then((responseObj) => {
+                    return responseObj;
+                });
+            });
+        }
+
+
+        /**
+         * Creates metadata on the artifact.
```

Comment:
I guess I don't really think it's necessary either... But it's kinda nice having a function explicitly for creating and one for updating?